### PR TITLE
Test index.js file

### PR DESCRIPTION
**Pull request by stephenplusplus**
When a user gets a localized gcloud instance:

``` js
var gcloud = require('gcloud')(/*…*/);
```

They have been missing out on subclass static properties:

``` js
gcloud.storage.acl // undefined
```

This fixes that and adds tests.

We should do a patch release after this lands.

### CODE REVIEW

**Comment by callmehiphop** [comment]
So what happens when a user creates a new API instance like..

```javascript
var gcloud = require('gcloud');
var search = gcloud.search({ ... });
```

Since search and pubsub are not taking advantage of the auto instantiation none of the static members are available through that method.

File: lib/index.js [hunk 1]
```diff
@@ -122,29 +124,39 @@ var util = require('./common/util.js');
...
     }
   };
+
+  // Extend each API to pick up static members, e.g. Datastore.int & Storage.acl
+  extend(api.bigquery, BigQuery);
+  extend(api.datastore, Datastore);
+  extend(api.dns, DNS);
+  extend(api.pubsub, PubSub);
+  extend(api.search, Search);
+  extend(api.storage, Storage);
```
Comment (callmehiphop) [inline lib/index.js]: This is maybe a little verbose? What if we did something like..

```javascript
function gcloud (config) {
  return {
    bigquery: createFactory(BigQuery, config),
    datastore: extend(new Datastore(config), Datastore),
    dns: createFactory(DNS, config),
    pubsub: createFactory(PubSub, config),
    search: createFactory(Search, config),
    storage: createFactory(Storage, config)
  };
}

function createFactory (ctor, config) {
  function factory (options) {
    return new ctor(util.extendGlobalConfig(config, options));
  }
  extend(factory, ctor);
  return factory;
}
```

Comment (stephenplusplus) [inline lib/index.js]: Check it out now and let me know if I went too crazy.

Comment (callmehiphop) [inline lib/index.js]: Check what out D:

Comment (stephenplusplus) [inline lib/index.js]: Lol, sorry. Check out #798.

**Pull request closed without merging**